Doc/.ruff.toml

```
@@ -0,0 +1,46 @@
+target-version = "py312"  # align with the version in oldest_supported_sphinx
+fix = true
+output-format = "full"
+extend-exclude = [
+    "includes/*",
+    "tools/extensions/c_annotations.py",
+    "tools/extensions/escape4chm.py",
+    "tools/extensions/patchlevel.py",
+    "tools/extensions/pyspecific.py",
+]
+
+[lint]
+preview = true
+select = [
+    "C4",  # flake8-comprehensions
+    "B",   # flake8-bugbear
+    "E",   # pycodestyle
+    "F",   # pyflakes
+    "FA",  # flake8-future-annotations
+    "FLY", # flynt
+    "FURB",# refurb
+    "G",   # flake8-logging-format
+    "I",   # isort
+    "LOG", # flake8-logging
+    "N",   # pep8-naming
+    "PERF",# perflint
+    "PGH", # pygrep-hooks
+    "PT",  # flake8-pytest-style
+    "TCH", # flake8-type-checking
+    "UP",  # pyupgrade
+    "W",   # pycodestyle
+]
+ignore = [
+    "E501",  # We use auto-formatting, so ignore line length errors
+]
+
+[format]
```

In other projects I generally have these settings enabled:

```suggestion
[format]
docstring-code-format = true
skip-magic-trailing-comma = true
```

(But I don't feel strongly about them)

---

I've enabled 'docstring-code-format'